src/status_emitter.rs

```diff
@@ -6,14 +6,34 @@ use std::{
     fmt::Debug,
     panic::RefUnwindSafe,
     path::{Path, PathBuf},
+    sync::{Arc, Mutex},
 };
-pub use text::*;
 pub mod debug;
-mod text;
 #[cfg(feature = "gha")]
 pub use gha::*;
 #[cfg(feature = "gha")]
 mod gha;
+pub use libtest_json::*;
+mod libtest_json;
+pub use text::*;
+mod text;
+
+// The testing framework employs the implementations of the various emitter traits
```

Comment:
turn this into module level docs

Reply:
I am back from vacation, and once again looking at the PR.

> turn this into module level docs

Will do!

Reply:
Done.